BlueDBOnDisk/src/main/java/org/bluedb/disk/file/FileUtils.java

```diff
@@ -176,6 +180,41 @@ public static void deleteIfExistsWithoutLock(Path path) throws BlueDbException {
 	public static DataOutputStream openDataOutputStream(File file) throws IOException {
 		return new DataOutputStream(new BufferedOutputStream(new FileOutputStream(file)));
 	}
+
+	protected static DataInputStream openDataInputStream(File file) throws IOException {
+		return new DataInputStream(new BufferedInputStream(new FileInputStream(file)));
+	}
+
+	/*
+	 * This is a copy of DataInputStream.readInt except that it returns null if the end of the file was reached
+	 * instead of throwing an exception. We noticed that reading through so many files in BlueDB was resulting
+	 * in TONS of EOFExceptions being thrown and caught which is a bit heavy. We could return an optional or
+	 * something but this is a really low level method that is going to be called a TON so I figured that
+	 * it is probably worth just handling a null return rather than creating a new object every time we
+	 * call it.
+	 */
+	public static Integer readInt(DataInputStream dataInputStream) throws IOException {
+		int ch1 = dataInputStream.read();
+		int ch2 = dataInputStream.read();
+		int ch3 = dataInputStream.read();
+		int ch4 = dataInputStream.read();
+		if ((ch1 | ch2 | ch3 | ch4) < 0) {
+			return null;
+		}
+		return ((ch1 << 24) + (ch2 << 16) + (ch3 << 8) + (ch4 << 0));
+	}
+
+	public static byte[] readAllBytes(InputStream is) throws IOException {
+		ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+		int numBytesRead;
+		byte[] nextBytes = new byte[4];

```

Comment:
Sure thing, if we want to refactor the buffer size into a constant somewhere we can, since it is only a singular usage I'll leave it for now though. I'll make that change though, that works for me 👍